components/Collections/src/MDCCollectionInfoBarView.m

```diff
@@ -0,0 +1,261 @@
+@end
+
+@implementation ShadowedView
++ (Class)layerClass {
+  return [MDCShadowLayer class];
+}
+@end
+
+@implementation MDCCollectionInfoBarView {
+  CGFloat _backgroundTransformY;
+  CALayer *_backgroundBorderLayer;
+  UITapGestureRecognizer *_tapGesture;
+}
+
+- (instancetype)initWithStyle:(MDCInfoBarStyle)style
+                         kind:(MDCInfoBarKind)kind
+               collectionView:(UICollectionView *)collectionView {
+  _style = style;
+  _kind = kind;
+
+  // Determine frame based on if this is a header or footer.
+  CGFloat offsetY = 0;
+  if (_kind == MDCInfoBarKindFooter) {
+    offsetY = collectionView.bounds.size.height - kInfoBarDefaultHeight;
+  }
+  CGRect frame = CGRectMake(0, offsetY, collectionView.bounds.size.width, kInfoBarDefaultHeight);
+
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCCollectionInfoBarViewInit];
+  }
+  return self;
+}
+
+- (void)commonMDCCollectionInfoBarViewInit {
+  self.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleTopMargin;
+  self.backgroundColor = [UIColor clearColor];
+  self.userInteractionEnabled = NO;
+
+  // Setup background view.
+  _backgroundView = [[ShadowedView alloc] initWithFrame:self.bounds];
+  _backgroundView.autoresizingMask = UIViewAutoresizingFlexibleWidth;
+  _backgroundView.clipsToBounds = YES;
+  _backgroundView.hidden = YES;
+  _backgroundTransformY =
+      (_kind == MDCInfoBarKindHeader) ? -kInfoBarDefaultHeight : kInfoBarDefaultHeight;
+  _backgroundView.transform = CGAffineTransformMakeTranslation(0, _backgroundTransformY);
+  _tapGesture =
+      [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleTapGesture:)];
+  [_backgroundView addGestureRecognizer:_tapGesture];
+  [self addSubview:_backgroundView];
+
+  // Setup label.
+  _titleLabel = [[UILabel alloc]
+      initWithFrame:CGRectInset(self.bounds, kInfoBarDefaultLabelHorizontalPadding, 0)];
+  _titleLabel.backgroundColor = [UIColor clearColor];
+  _titleLabel.textAlignment = NSTextAlignmentCenter;
+  _titleLabel.font = [MDCTypography body1Font];
+  _titleLabel.autoresizingMask = UIViewAutoresizingFlexibleWidth;
+  [_backgroundView addSubview:_titleLabel];
+
+  [self prepareLayout];
+}
+
+- (void)prepareLayout {
+  // Set info bar defaults based on style.
+  if (_style == MDCInfoBarStyleHUD) {
+    _allowsTap = NO;
+    _textAlignment = NSTextAlignmentLeft;
+    _tintColor = ColorFromRGB(kInfoBarBlueColor);
+    _titleColor = [UIColor whiteColor];
+    _autoDismissAfterDuration = kInfoBarDefaultDismissal;
+    _backgroundView.alpha = 0.9f;
+    self.shouldApplyBorder = NO;
+    self.isAccessibilityElement = NO;
+    self.accessibilityTraits = UIAccessibilityTraitNone;
+    self.accessibilityLabel = nil;
+  } else if (_style == MDCInfoBarStyleActionable) {
+    _allowsTap = YES;
+    _textAlignment = NSTextAlignmentCenter;
+    _tintColor = [UIColor whiteColor];
+    _titleColor = ColorFromRGB(kInfoBarRedColor);
```

Comment:
I would expect this and the other visual properties to be set in the +initialize on the UIAppearance proxy.
[MDCInfoBar appearance].titleColor = RGB(DefaultTitleColor);